ballerina/resiliency_failover_client.bal

```diff
@@ -99,14 +99,16 @@ public client isolated class FailoverClient {
     } external;
 
     private isolated function processPost(string path, RequestMessage message, TargetType targetType, 
-            string? mediaType, map<string|string[]>? headers) returns Response|anydata|ClientError {
+            string? mediaType, map<string|string[]>? headers) returns Response|StatusCodeResponse|anydata|ClientError {
         Request req = check buildRequest(message, mediaType);
         populateOptions(req, mediaType, headers);
         var result = self.performFailoverAction(path, req, HTTP_POST);
         if result is HttpFuture {
             return getInvalidTypeError();
-        } else if result is Response || result is ClientError {
-            return processResponse(result, targetType, self.requireValidation);
+        } else if result is ClientError {
+            return result;
+        } else if result is Response {
+            return externProcessResponse(result, targetType, self.requireValidation);
         } else {
             panic error ClientError("invalid response type received");
         }
```

Comment:
I think this complete code block could be moved into `processResponse` function.

Reply:
Fixed via [bce164f](https://github.com/ballerina-platform/module-ballerina-http/pull/1915/commits/bce164f18e1e33b548e2db2aa20f294e437a2202)